backend/app/readers/readers_instantiator.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from app.models.settings.settings import settings
+from app.readers.reader.reader import Reader
+from app.readers.reader.wikipedia_reader import WikipediaReader
+
+
+class ReadersInstantiator:
+    """
+    An Instantiator for the different Readers.
+    """
+
+    def __init__(self) -> None:
+        self.__record_types: frozenset[str] | None = self.__get_record_types()
+        self.__file_output_paths: list[Path] | None = self.__get_output_paths()
+
+    def __get_record_types(self) -> frozenset[str] | None:
```

Comment:
You don't really need this if settings.record_types is always a set. (Can use a Pydantic validator to transform it from a str setting to a frozenset.)

Reply:
Then you don't need this method at all.